ipa-core/src/secret_sharing/replicated/semi_honest/additive_share.rs

```diff
@@ -255,6 +255,7 @@ where
     <S as ArrayAccess>::Output: SharedValue,
 {
     type Output = AdditiveShare<<S as ArrayAccess>::Output>;
+    type Iter<'a> = ASIterator<S::Iter<'a>>;
```

Comment:
I love this. Why didn't we consider this earlier when we ran into the issue with trait bounds for references (i.e when I tried to require Into_Iterator for a reference to S)?

Reply:
I think the combined effort to push boolean array and vectorization added some generic traits like `ArrayAccess`, `StdArray` that we can now leverage to provide a generic `iter` implementation for all shares over values that implement `ArrayAccess`